sale_automatic_workflow/models/automatic_workflow_job.py

```diff
@@ -140,6 +140,35 @@ def _sale_done(self, sale_done_filter):
             with savepoint(self.env.cr), force_company(self.env, sale.company_id):
                 self._do_sale_done(sale, sale_done_filter)
 
+    def _prepare_dict_account_payment(self, invoice):
+        partner_type = (
+            invoice.type in ("out_invoice", "out_refund") and "customer" or "supplier"
+        )
+        return {
+            "invoice_ids": [(6, 0, invoice.ids)],
+            "amount": invoice.residual,
+            "payment_date": fields.Date.context_today(self),
+            "communication": invoice.reference or invoice.number,
```

Comment:
`invoice.number` doesn't exist anymore. It should be `invoice.name` for customer invoices, and `invoice.reference` for vendor bills.